### PR TITLE
Notify 445

### PR DESCRIPTION
**Pull request by A-Shumway42**
Added:

- Daily message limit for service with amount.
Screen Shot 2023-06-01 at 9 42 08 AM

- Remaining daily message amount displayed before csv upload as well as after with remaining amount if csv is sent.
Screen Shot 2023-06-01 at 9 44 18 AM

- Daily message limit added to side nav with format "daily remaining messages / message limit" and global daily remaining messages across services.
Screen Shot 2023-05-31 at 12 32 45 PM

It's worth noting that I added a global variable to the `config` in the admin repo for the `daily limit across all services of 5000`. We can't access the `config` in the api repo. Steven mentioned it would be better at some point to add that to the utils repo so both the api and ui can access it instead of hard coding it in both locations.

### CODE REVIEW

**Comment by tdlowden** [comment]
#445

File: app/main/views/send.py
Comment (ccostino) [inline app/main/views/send.py:84]: I don't think there's a need for the parentheses here; in this case they're likely not hurting anything, but in other contexts what you might end up doing is creating a [`tuple`](https://docs.python.org/3/library/stdtypes.html#tuple)

Again, in this case you're okay because we're just dealing with an arithmetic operation, though the parentheses are superfluous because they're not doing anything for us.  We don't have an order of operations situation we need to be accounting for, meaning:

```python
remaining_messages = (current_service.message_limit - notification_count)
```

is the same as:

```python
remaining_messages = current_service.message_limit - notification_count
```

...which is more readable in this case. :-)

I bring this up though because when I initially saw the parentheses they gave me pause because I couldn't remember if that'd create a tuple or not.  In case you're not familiar with tuples, they are like Python lists but with two important distinctions:

- The syntax uses `(` and `)` instead of `[` and `]`
- Tuples are immutable

Here are a couple of quick examples of tuples:

```python
# Creating an empty tuple.
empty_tuple = ()

# Creating a tuple with some elements in it.  Note the trailing comma (which is also done with regular lists).
new_tuple = ('here', 'is', 'a', 'new', 'tuple',)

# Converting a sequence (e.g., a list) to a tuple.
new_list = ['here', 'is', 'a', 'sequence',]
converted_tuple = tuple(new_list)
```

In any case, all that to say, we're okay without the parentheses here!

```suggestion
    remaining_messages = current_service.message_limit - notification_count
```

File: app/notify_client/service_api_client.py
Comment (ccostino) [inline app/notify_client/service_api_client.py:540]: Small styling nit-pick:  no need for the extra line here.

```suggestion
```

File: tests/app/main/views/test_send.py
Comment (ccostino) [inline tests/app/main/views/test_send.py:1754]: Ah, is this switch because of the multiple context processors now being called?  Or was there something else that required the change?

**Review by ccostino — COMMENTED** [review]
Thanks, @A-Shumway42!

There were a couple of small things I saw that I offered suggestions for with additional context.  I also had a quick question about the change to the test that I wasn't sure about.

Everything else looks good!

File: app/__init__.py
Comment (A-Shumway42) [reply app/__init__.py:318]: I might have rollbacked there on accident. I'll change that.

File: app/notify_client/service_api_client.py
Comment (A-Shumway42) [reply app/notify_client/service_api_client.py:540]: I removed the empty white space as well as the one in the function above.

File: tests/app/main/views/test_send.py
Comment (A-Shumway42) [reply tests/app/main/views/test_send.py:1754]: This switch was due to the get_notification_count() being called twice in the send workflow in `send.py`. Would there be a better way of doing it?

File: app/__init__.py
Comment (A-Shumway42) [reply app/__init__.py]: There looks to be an error in github actions after this change. It's references the object current_service isn't available? It looks like it's some form issue where the error is occuring.

File: app/config.py
Comment (stvnrlly) [inline app/config.py:50]: Ah, I see that this is necessary for calculating the remaining amount, not just showing the limit. Let's open an issue for `notifications-utils` to move the limit there, and then we can reference that in both applications without needing to maintain them in parallel.

**Review by stvnrlly — APPROVED** [review]
Looks good, but do a little more work after merging this to avoid having our global limit set independently in both applications.

File: app/config.py
Comment (ccostino) [reply app/config.py:50]: 👍 @stvnrlly!

@A-Shumway42 and I discussed the same thing when pairing through this, and I was suspecting that having it be managed and accessible in `notifications-utils` would make the most sense rather than having it hardcoded in multiple spots.

I agree:  finish this up and merge it, then create a separate issue for that work and follow up in here again once `notification-utils` has full support for it.

File: app/__init__.py
Comment (ccostino) [reply app/__init__.py]: 🤔 maybe we do need to keep it then...  I'll take another look just to be sure and let you know.

File: tests/app/main/views/test_send.py
Comment (ccostino) [reply tests/app/main/views/test_send.py:1754]: No, I think you're okay!  I just wanted to make sure I understood what was happening and my assumption was correct. :-) Thanks for explaining!

File: app/__init__.py
Comment (ccostino) [reply app/__init__.py]: Okay yeah @A-Shumway42, I'm sorry about that - I thought we were in the clear!  Keep the change as it was with `hasattr` for now.  This is going to take some additional investigation to see how the `current_service` is populated and when.

We'll figure it out!  No need to hold up this PR for it, though. :-)

**Pull request merged**
